_posts/2022-06-08-packaging-functions-with-helm.md

```diff
@@ -0,0 +1,221 @@
+```
+.
+└── functions
+    ├── charts
+    ├── Chart.yaml
+    ├── templates
+    │   ├── deployment.yaml
+    │   ├── _helpers.tpl
+    │   ├── hpa.yaml
+    │   ├── ingress.yaml
+    │   ├── NOTES.txt
+    │   ├── serviceaccount.yaml
+    │   ├── service.yaml
+    │   └── tests
+    │       └── test-connection.yaml
+    └── values.yaml
+```
+
+Not all files generated by `helm create` are required to create a deployable Chart. At a minimum Helm expects the following files:
+- Chart.yaml: This file contains metadata about the chart (e.g. version, author, etc)
+- values.yaml: The `values.yaml` file contains default values for the templated parts of the Chart. These values can be set while deploying the Chart using the `--values` or `--set` flags.
+- template files: The templates folder contains all the Kubernetes manifest templates that define your application. This is where we will place our function CRDs.
+
+The optional `NOTES.txt` file is a plain text file that can be used to define a message or some usage notes. This message will be printed to the console when the chart is deployed. The templates folder will also contain some examples of different Kubernetes manifest files. They can be safely removed since we will replace them with our own function templates.
+
+### Templating functions
+We will use the `faas-cli generate` command to create the function custom resource for the functions defined in the `stack.yaml` file. The output of the generate command will be written to a file in the templates directory.
+```
+faas-cli generate -f marketing-list-func.yaml > chart/functions/templates/marketing-list-func.yaml
+```
+
+Once the function CRDs are generated you can start editing them.
+We will modify the `marketing-list-func.yaml` template to make the image configurable.
+
+{% raw %}
+```yaml
+---
+apiVersion: openfaas.com/v1
+kind: Function
+metadata:
+  name: marketing-list
+  namespace: openfaas-fn
+spec:
+  name: marketing-list
+  image: {{ .Values.marketingList.image }}
+```
+{% endraw %}
+
+Edit the `values.yaml` file to set the default value for the image:
+```
+marketingList:
+  image: docker.io/functions/marketing-list:0.1.0
+``` 
+
+### Installing the chart
+We can deploy our functions by running `helm install`. Make sure your function images are available before deploying them. The Helm Chart contains the deployment configuration for the functions but no build instructions. Building functions still has to be done using the faas-cli or another build method.
+
+```
+helm install functions chart/functions
+```
+
+When the installation is successful, helm will print out some information about the deployment along with the messages from the `NOTES.txt` file if it was included in the Chart.
+```
+NAME: functions
+LAST DEPLOYED: Wed May 18 10:55:53 2022
+NAMESPACE: default
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
+```
+
+### Deploying a new version of a function
```

Comment:
For each heading where you have an "-ing" as the first word, please change this to the imperative tense:

deploying => deploy
generating => generate